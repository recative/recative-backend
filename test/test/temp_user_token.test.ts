import { describe, expect, it } from "vitest";

import { SudoFetcher } from "../src/fetcher/authFetcher";

describe("admin token crud", () => {
    it('create admin token', async () => {
        let date = new Date()
        // add 5 days
        date.setDate(date.getDate() + 5)
        const resp = await SudoFetcher.path('/admin/token').method('post').create()({
            is_valid:true,
            comment:"a token for test",
            admin_permission: ["write"],
            token: undefined,
            expired_at: date.toISOString(),
        })

        expect (resp.ok).toBe(true)
        expect (resp.data).matchSnapshot()
    });
});