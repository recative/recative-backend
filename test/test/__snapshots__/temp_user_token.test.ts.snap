// Vitest Snapshot v1

exports[`admin token crud > create admin token 1`] = `
{
  "token": "d9b2e38c-3e49-45b5-b727-3af85f1f80f9",
}
`;
